bin/sessionmirror.py

```diff
@@ -0,0 +1,749 @@
+    except IOError as error:
+        print('ERROR:writing xml file: {}: {}'.format(file_path, str(error)))
+
+
+def check_attributes(src_obj, dest_obj, dtype=None):
+    '''Check that attributes on dest match those on src'''
+
+    if dtype is None:
+        dtype = src_obj.datatype()
+
+    if dtype == 'xnat:projectData':
+        attr_list = PROJ_ATTRS
+    elif dtype == 'xnat:subjectData':
+        attr_list = SUBJ_ATTRS
+    elif dtype == 'xnat:mrSessionData':
+        attr_list = MR_EXP_ATTRS
+    elif dtype == 'xnat:mrScanData':
+        attr_list = MR_SCAN_ATTRS
+    elif dtype == 'xnat:scScanData':
+        attr_list = SC_SCAN_ATTRS
+    elif dtype == 'xnat:petScanData':
+        attr_list = PET_SCAN_ATTRS
+    elif dtype == 'xnat:ctScanData':
+        attr_list = CT_SCAN_ATTRS
+    elif dtype == 'proc:genProcData':
+        attr_list = PROC_ATTRS
+    elif dtype == 'xnat:otherDicomScanData':
+        attr_list = OTHER_DICOM_SCAN_ATTRS
+    else:
+        print('WARN:Unknown Type:%s' % dtype)
+        return
+
+    for a in attr_list:
+        src_v = src_obj.attrs.get(a)
+        src_v = src_v.replace("\\", "|")
+        dest_v = dest_obj.attrs.get(a)
+        if src_v != dest_v:
+            print('WARN: Attri mismatch, setting again:%s:src_v=%s,dest_v=%s'
+                  % (a, src_v, dest_v))
+            dest_obj.attrs.set(a, src_v)
+
+
+def create_parser():
+    import argparse
+    """Parse commandline arguments."""
+    parser = argparse.ArgumentParser(description='Downloads a given experiment/'\
+        'session from an XNAT instance and uploads it to an independent one. '\
+        'Only DICOM resources will be imported. \n\n'\
+        'More information at: https://wiki.xnat.org/docs16/4-developer-documen'\
+        'tation/xnat-rest-api-directory/data-services-import',
+        formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument('--h1', '--source_config', dest='source_config',
+        help='Source XNAT configuration file', required=True)
+    parser.add_argument('--h2', '--dest_config', dest='dest_config',
+        help='Destination XNAT configuration file', required=True)
+    parser.add_argument('-e','--experiment_id',
+        help='Which resource to download? (Entity name/identifier)', required=True)
+    parser.add_argument('-p','--project_id', dest='project_id',
+        help='Which project to store the resource in', required=True)
+
+    parser.add_argument('-v','--verbose', dest='verbose', action='store_true',
+        default=False, help='Display verbosal information (optional)',
+        required=False)
+
+    return parser
+
+
+
+def main(args):
+    x1 = pyxnat.Interface(config=args.source_config)
+    x2 = pyxnat.Interface(config=args.dest_config)
+
+    columns = ['subject_label', 'label']
+    e1 = x1.array.experiments(experiment_id=args.experiment_id,
+        columns=columns).data[0]
+    p = x2.select.project(args.project_id)
+    s = p.subject(e1['subject_label'])
```

Comment:
what if subject exists?

Reply:
> as long as experiment does not exist
Experiment never exists since it is created/duplicated.

> what if subject exists?
After a thorough analysis of the code (sic), it appears that if subject exists, nothing happens.

Reply:
```suggestion
    s = p.subject(e1['subject_label'])
    if not s.exists():
        s.create()
```